radixdlt-core/radixdlt/src/integration/java/com/radixdlt/integration/api/actors/actions/CreateTokenDefinition.java

```diff
@@ -62,57 +61,108 @@
  * permissions under this License.
  */
 
-package com.radixdlt.api.data.action;
+package com.radixdlt.integration.api.actors.actions;
 
-import com.radixdlt.atom.TxAction;
-import com.radixdlt.atom.actions.CreateFixedToken;
-import com.radixdlt.identifiers.REAddr;
+import com.radixdlt.api.core.openapitools.model.ConstructionDeriveRequestMetadata;
+import com.radixdlt.api.core.openapitools.model.ConstructionDeriveRequestMetadataToken;
+import com.radixdlt.api.core.openapitools.model.Data;
+import com.radixdlt.api.core.openapitools.model.EngineConfiguration;
+import com.radixdlt.api.core.openapitools.model.EntityIdentifier;
+import com.radixdlt.api.core.openapitools.model.Operation;
+import com.radixdlt.api.core.openapitools.model.OperationGroup;
+import com.radixdlt.api.core.openapitools.model.PreparedValidatorRegistered;
+import com.radixdlt.api.core.openapitools.model.ResourceAmount;
+import com.radixdlt.api.core.openapitools.model.TokenData;
+import com.radixdlt.api.core.openapitools.model.TokenMetadata;
+import com.radixdlt.api.core.openapitools.model.TokenResourceIdentifier;
 import com.radixdlt.utils.UInt256;
 
-import java.util.stream.Stream;
+import java.util.List;
+import java.util.function.Function;
 
-class CreateFixedTokenAction implements TransactionAction, ResourceAction {
-	private final REAddr from;
-	private final UInt256 amount;
-	private final REAddr rri;
-	private final String name;
+public final class CreateTokenDefinition implements NodeTransactionAction {
 	private final String symbol;
-	private final String iconUrl;
-	private final String tokenUrl;
-	private final String description;
+	private final UInt256 amount;
+	private final EntityIdentifier owner;
+	private final EntityIdentifier to;
 
-	CreateFixedTokenAction(
-		REAddr from,
-		UInt256 amount,
-		REAddr rri,
-		String name,
-		String symbol,
-		String iconUrl,
-		String tokenUrl,
-		String description
-	) {
-		this.from = from;
-		this.amount = amount;
-		this.rri = rri;
-		this.name = name;
+	private CreateTokenDefinition(String symbol, UInt256 amount, EntityIdentifier owner, EntityIdentifier to) {
 		this.symbol = symbol;
-		this.iconUrl = iconUrl;
-		this.tokenUrl = tokenUrl;
-		this.description = description;
+		this.amount = amount;
+		this.owner = owner;
+		this.to = to;
 	}
 
-	@Override
-	public Stream<TxAction> toAction() {
-		return Stream.of(new CreateFixedToken(TransactionAction.rriValue(rri), from, symbol, name, description, iconUrl, tokenUrl, amount));
+	public static CreateTokenDefinition fixedTokenSupply(String symbol, UInt256 amount, EntityIdentifier to) {
+		return new CreateTokenDefinition(symbol, amount, null, to);
```

Comment:
Do you think we could avoid the use of nulls?

Reply:
It's use is internal to the class so not a big deal imo.